common/singleChunkReader.go

```diff
@@ -412,6 +412,13 @@ func (cr *singleChunkReader) Close() error {
 	// do the real work
 	cr.closeBuffer()
 	cr.isClosed = true
+
+	/*
+	 * Set chunkLogger to nil, so that chunkStatusLogger be GCed. Need to revisit this as
+	 * should not required to set nil. But we are seeing this never chunkstatus logger memory not freed.
+	 * It might be reference of cr held by someone.
+	 */
```

Comment:
let's have a TODO to attract attention. I've rephrased it too, pls consider
	/*
	 * Set chunkLogger to nil, so that chunkStatusLogger can be GC'ed. 
	 * 
	 * TODO: We should not need to explicitly set this to nil but today we have a yet-unknown ref on cr which
	 *       is leaking this "big" chunkStatusLogger memory, so we cause that to be freed by force dropping this ref.
	 */

Reply:
Btw is there a solid guarantee that cr.chunkLogger won't be used after it's set to nil.
Can we add check for cr.chunkLogger != nil to blockingPrefetch where it's being used?